web/src/test/scala/quasar/api/services/DataServiceSpec.scala

```diff
@@ -0,0 +1,511 @@
+      "be 400 for missing Destination header" ! prop { path: Path[Abs,File,Sandboxed] =>
+        val pathString = posixCodec.printPath(path)
+        val request = Request(uri = Uri(path = pathString), method = Method.MOVE)
+        val response = service(emptyMem)(request).run
+        response.status must_== Status.BadRequest
+        response.as[String].run must_== "The 'Destination' header must be specified"
+      }
+      "be 404 for missing source file" ! prop { (source: Path[Abs,File,Sandboxed], destination: Path[Abs,File,Sandboxed]) =>
+        val sourceString = posixCodec.printPath(source)
+        val destinationString = posixCodec.printPath(destination)
+        val request = Request(uri = Uri(path = sourceString), headers = Headers(Header("Destination", destinationString)), method = Method.MOVE)
+        val response = service(emptyMem)(request).run
+        response.status must_== Status.NotFound
+        response.as[String].run must_== s"$sourceString: doesn't exist"
+      }
+      "be 404 for missing destination file" ! prop { (source: Path[Abs,File,Sandboxed], destination: Path[Abs,File,Sandboxed]) =>
+        todo // Figure out what we are doing with nested backends because not sure this test makes sense without them
+      }
+      "be 201 with file" ! prop {(source: Path[Abs,File,Sandboxed], destination: Path[Abs,File,Sandboxed], data: Vector[Data]) =>
+        val sourceString = posixCodec.printPath(source)
+        val destinationString = posixCodec.printPath(destination)
+        val request = Request(uri = Uri(path = sourceString), headers = Headers(Header("Destination", destinationString)), method = Method.MOVE)
+        val filesystem = InMemState.fromFiles(Map(source -> data))
+        val response = service(filesystem)(request).run
+        response.status must_== Status.Created
+        // TODO: Check that the file was moved properly (not in old test though...)
+      }
+      "be 201 with dir" ! prop {(sourceDir: Path[Abs,Dir,Sandboxed], sampleFiles: Map[String, Vector[Data]], destinationDir: Path[Abs,Dir,Sandboxed]) =>
+        val sourceDirString = posixCodec.printPath(sourceDir)
+        val fileMapping = sampleFiles.map{ case (fileName,data) => (sourceDir </> file(fileName), data)}
+        val destinationString = posixCodec.printPath(destinationDir)
+        val request = Request(uri = Uri(path = sourceDirString), headers = Headers(Header("Destination", destinationString)), method = Method.MOVE)
+        val filesystem = InMemState fromFiles fileMapping
+        val response = service(filesystem)(request).run
+        response.status must_== Status.Created
+        // TODO: Check that the file was moved properly (not in old test though...)
+      }
+      "be 501 for src and dst not in the same backend" >> todo // What are we doing? make sure that we test all error conditions that the backend can return
+    }
+    "DELETE" >> {
+      "be 404 for missing backend" >> todo // What are we doing?
```

Comment:
I think we can drop this one as the nested interpreter will respond with a "path not found" or equivalent if there isn't a backend for a path.